op-acceptance-tests/tests/interop/loadtest/interop_load_test.go

```diff
@@ -1,139 +1,109 @@
-			execMsgsTx.PlannedTx.Nonce.DependOn(prevNonce)
-			execMsgsTx.PlannedTx.Nonce.Fn(func(_ context.Context) (uint64, error) {
-				prevNonceU64, err := prevNonce.Get()
-				if err != nil {
-					return 0, err
-				}
-				return prevNonceU64 + 1, nil
-			})
-		}
-		execMsgsTxs = append(execMsgsTxs, execMsgsTx)
-	}
 	var wg sync.WaitGroup
-	wg.Add(len(execMsgsTxs))
-	for _, execMsgsTx := range execMsgsTxs {
-		go func(tx *txplan.PlannedTx) {
-			defer wg.Done()
-			receipt, err := tx.Included.Eval(t.Ctx())
-			t.Require().NoError(err)
-			t.Require().Len(receipt.Logs, numMsgs)
-			_, err = tx.Success.Eval(t.Ctx())
-			t.Require().NoError(err)
+	defer wg.Wait()
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		SpamInteropTxs(t, &wg, numInitTxs, L2A, L2B, sys.Supervisor)
+	}()
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		SpamInteropTxs(t, &wg, numInitTxs, L2B, L2A, sys.Supervisor)
+	}()
+}
+
+func SpamInteropTxs(t devtest.T, wg *sync.WaitGroup, numInitTxs uint64, source *L2, dest *L2, supervisor *dsl.Supervisor) {
+	msgsCh := make(chan []types.Message, 100)
+	defer close(msgsCh)
 
-			// Wait for the transaction to be cross-safe.
-			includedBlock, err := tx.IncludedBlock.Eval(t.Ctx())
-			t.Require().NoError(err)
-			for {
-				// NOTE: it may be desirable to query proxyd instead of the supervisor if/when the devstack supports it.
-				crossSafeID, err := sys.Supervisor.Escape().QueryAPI().CrossSafe(t.Ctx(), l2ELB.ChainID())
-				t.Require().NoError(err)
-				if includedBlock.ID().Number <= crossSafeID.Derived.Number {
-					break
-				}
-				l2ELB.WaitForBlock()
+	// Spam executing messages.
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		executors := []Executor{
+			NewValidExecutor(dest.Funder, dest.EL, supervisor),
+			NewDelayedExecutor(NewValidExecutor(dest.Funder, dest.EL, supervisor), wg, time.Minute),
+			NewInvalidExecutor(dest.Funder, dest.EL, makeInvalidChainID),
+			NewInvalidExecutor(dest.Funder, dest.EL, makeInvalidBlockNumber),
+			NewInvalidExecutor(dest.Funder, dest.EL, makeInvalidLogIndex),
+			NewInvalidExecutor(dest.Funder, dest.EL, makeInvalidOrigin),
+			NewInvalidExecutor(dest.Funder, dest.EL, makeInvalidPayloadHash),
+			NewInvalidExecutor(dest.Funder, dest.EL, makeInvalidTimestamp),
+		}
+		for msgs := range msgsCh {
+			for _, executor := range executors {
+				wg.Add(1)
+				go func(executor Executor, msgs []types.Message) {
+					defer wg.Done()
+					executor.Execute(t, msgs)
+				}(executor, msgs)
 			}
-			// Sanity check that includedBlock is still in the canonical chain.
-			_, err = l2ELB.Escape().EthClient().BlockRefByHash(t.Ctx(), includedBlock.Hash)
-			t.Require().NoError(err)
-		}(execMsgsTx.PlannedTx)
+		}
+	}()
+
+	// Spam initiating messages.
+	eventLogger := source.Funder.NewFundedEOA(eth.OneEther).DeployEventLogger()
+	initiators := []Initiator{
+		NewManyMsgsInitiator(source.Funder, source.EL, eventLogger),
+		NewLargeMsgInitiator(source.Funder, source.EL, eventLogger),
```

Comment:
That this test only originate txs a single EOA may hinder it's performance because the mempool may have to do more work to ensure that transactions are ordered correctly according to nonce, preventing as many txs from being included in the block at the extreme.

Reply:
Ah good observation, we will need to evade op-geth's mempool DoS protection. op-geth [limits](https://github.com/ethereum-optimism/op-geth/blob/68075997f33907401a93216aa426514c5ddc8870/core/txpool/legacypool/legacypool.go#L1563) the number of txs per account, so we should be using new EOAs. I'll work on this tomorrow.